### DISCUSSION TITLE
[ideas] Fail earlier for bad filenames (probably only an issue in Windows)

### DISCUSSION THREAD

**Discussion by billdenney** [ropensci/targets#1125] [Ideas]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

As has happened for me several times over time, I have Windows filename-length issues with some of my targets.  Would it be possible to have targets fail faster if there will be a long filename issue so that a long run would not start instead of failing at the end?

The reprex below should fail on Windows due to a long filename, and the request is that it fails at the beginning of the 30 seconds rather than at the end.  In my real use case, it was a computational model that took about 10 hours to run.

One way that this could happen is to just try touching the scratch and/or objects file just before the target is started.

```
library(targets)

list(
  tar_target(
    two_123456789101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899100101102103104105106107108109110111112113114115116117118119120121122123124125126127128129130131132133134135136137138139140141142143144145146147148149150151152153154155156157158159160161162163164165166167168169170171172173174175176177178179180181182183184185186187188189190191192193194195196197198199200201202203204205206207208209210211212213214215216217218219220221222223224225226227228229230231232233234235236237238239240241242243244245246247248249250251252253254255256257258259260261262263264265266267268269270271272273274275276277278279280281282283284285286287288289290291292293294295296297298299300301302303304305306307308309310311312313314315316317318319320321322323324325326327328329330331332333334335336337338339340341342343344345346347348349350351352353354355356357358359360361362363364365366367368369370371372373374375376377378379380381382383384385386387388389390391392393394395396397398399400401402403404405406407408409410411412413414415416417418419420421422423424425426427428429430431432433434435436437438439440441442443444445446447448449450451452453454455456457458459460461462463464465466467468469470471472473474475476477478479480481482483484485486487488489490491492493494495496497498499500,
    {
      Sys.sleep(30)
      2
    }
  )
)
```

**Comment by billdenney** [comment]
I just found that this should have been resolved with R 4.3.0 (which I'm using): https://cran.r-project.org/doc/manuals/r-release/NEWS.html and specifically https://blog.r-project.org/2023/03/07/path-length-limit-on-windows/

So, maybe this could be a fix upstream in gzfile or other functions needed for file interaction.

**Comment by wlandau** [comment]
As with `drake`, I have no plans to implement special fixes in `targets`. You can control the location of the data store using the `store` argument or `tar_config_set(store = "...")`, or move your whole project higher in the directory tree.
